test/symbol_name/nmcheck_prefix.pl

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/perl
+
+# Copyright (c) 2017      IBM Corporation. All rights reserved.
+
+# env var MYBASE should be the top dir where ompi/ opal/ orte/ test/ etc live.
+# env var OMPI_LIBMPI_NAME should be @OMPI_LIBMPI_NAME@ from automake
+#
+# Most likely the libs we want to check are
+#     ompi/.libs/lib<mpi>.so
+#     ompi/mpi/fortran/mpif-h/.libs/lib<mpi>_mpifh.so
+#     ompi/mpi/fortran/use-mpi-tkr/.libs/lib<mpi>_usempi.so
+#     orte/.libs/libopen-rte.so
+#     opal/.libs/libopen-pal.so
+# but I hate to assume those are the locations, so I'll use 'find' and
+# test whatever ".so"s are found.
+
+@libs = ();
+$mpi = $ENV{OMPI_LIBMPI_NAME};
+for $name (
+    "lib${mpi}.so",
+    "lib${mpi}_mpifh.so",
+    "lib${mpi}_usempi.so",
+    "libopen-rte.so",
+    "libopen-pal.so" )
+{
+    for $loc (split(/\n/, `find $ENV{MYBASE} -name $name`)) {
+        push @libs, $loc;
+    }
+}
+
+print "Checking for bad symbol names in the main libs:\n";
+$isbad = 0;
+for $lib (@libs) {
+    print "checking $lib\n";
+    check_lib_for_bad_exports($lib);
+}
+if ($isbad) { exit(-1); }
+exit(0);
+
+
+sub check_lib_for_bad_exports {
+    my $lib = $_[0];
+    my @symbols;
+    my $s;
+
+    @symbols = get_nm($lib, 'all');
+
+    # grep to get rid of symbol prefixes that are considered acceptable,
+    # leaving behind anything bad:
+    @symbols = grep(!/^ompi_/i, @symbols);
+    @symbols = grep(!/^opal_/i, @symbols);
+    @symbols = grep(!/^orte_/i, @symbols);
+    @symbols = grep(!/^orted_/i, @symbols);
+    @symbols = grep(!/^mpi_/i, @symbols);
+    @symbols = grep(!/^pmpi_/i, @symbols);
+    @symbols = grep(!/^pmix_/i, @symbols);
+    @symbols = grep(!/^mpit_/, @symbols);
+    @symbols = grep(!/^ompit_/, @symbols);
```

Comment:
Oh, and I guess some of these are MPI_T_ related as well. So we could claim mpit_* for that, or name them mpi_t_*. I don't really like how mpi_t_* looks like a fortran entrypoint, I'd kind of prefer all mpi_* be actual MPI calls. But all of MPI_T_ starts with MPI_ already...